.github/actions/setup.yml

```diff
@@ -0,0 +1,11 @@
+name: Setup
```

Comment:
@baileympearson for context Daria asked if we could use [composite actions](https://wallis.dev/blog/composite-github-actions) to keep some shared steps consistent, if you look at the runs before the latest commit it's unable to find the file. Not sure why, but I believe it might have to exist on main before we can use it. So I made all the actions have the same block of setup and after merging we should be able to switch them all to this file. 

We also discussed using this for the publish step but a composite action does not have access to the repo secrets that provenance needs to work.